_posts/2016-09-16-why-linting-errors-should-cause-tests-to-fail.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Why linting errors should cause tests to fail
+layout: post
+summary: Making the case against allowing linting errors in master.
+---
+
+I recently had a discussion with a colleague who argued that:
+
+> Linting errors should not block a pull request from being merged.
+
+I found myself disagreeing vehemently but unable to clearly state why. I've
+written this post to help clarify my thoughts on the position.
+
+In this post I will attempt to convince you that a linting error should be
+treated as equivalent to test failure, and should not be permitted in your
+production branch.
+
+## Some specifics
+
+At [Hearsay Social](https://hearsaysocial.com/company/careers/engineering/) we
+use [GitHub pull requests](https://help.github.com/articles/about-pull-requests/)
+to manage changes to our code base. When a pull request is opened, [Travis
+CI](https://travis-ci.org) picks it up, and runs our test suite. The two
+requirements for merging a pull request into master, our production branch,
+are:
+
+* Approval from one other engineer
+* Passing Travis tests
+
+We recently configured our Travis tests to include running
+[ESLint](http://eslint.org/). This means that code that does not conform to our
+linting standards cannot be merged into master. While my colleague and I were
+both in agreement that linting is valuable, it was this change that sparked
+debate.
+
+## The simplified argument
+
+My colleague's main point against including ESLint in Travis was that, in
+a case where the site is down and we need to ship a hotfix, it would be
+terrible for a misaligned curly brace to prevent merging the fix and thus
+prolong our down-time.
+
+I must admit that I agree with his premise. It is far more important to bring
+the site back up than for master to be free of linting errors. However,
+I disagree that removing linting from our test suite is the best way to address
+this concern.
+
+I would argue that in the case of a site-down hotfix, you should have
+a mechanism for bypassing tests all together. There are many types of  of ways
+in which tests test failures which should not be a blocker to shipping
+a hotfix. For example:
+
+* Transient test failures cause by not properly mocking out the system clock
+* Test failures introduced by your package repository being down.
```

Comment:
No, not a better name. Maybe just put the examples in a paranthetical? Like: `... introduced by your package repository being down (e.g. Pypi, Packagist, npmjs.org, rubygems.org)`

Could just be me that's confused or misread this, in which case, disregard.